spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/ParameterSynchronizer.scala

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.intel.analytics.bigdl.utils
+
+import java.util.concurrent.{ConcurrentHashMap, CyclicBarrier}
+
+import com.intel.analytics.bigdl.tensor.Tensor
+import org.apache.spark.TaskContext
+
+import scala.reflect._
+
+/**
+ * A parameter synchronizer among threads per task
+ */
+object ParameterSynchronizer {
+
+  val fEvents = new java.util.concurrent.ConcurrentHashMap[String, Event[Float]]
```

Comment:
why there're two kind of events?

Reply:
For different types, as `SpatialBatchNormalization` implements explicitly type specific methods, this way would reduce the implicit conversion cost